### PR TITLE
Port Austin's web_tests tests

### PR DESCRIPTION
**Pull request by kainino0x**

### CODE REVIEW

File: src/framework/fixture.ts [hunk 1]
```diff
@@ -4,9 +4,10 @@ import { ParamsAny } from './params/index.js';
 // A Fixture is a class used to instantiate each test case at run time.
 // A new instance of the Fixture is created for every single test case
 // (i.e. every time the test function is run).
-export abstract class Fixture {
+export class Fixture {
   params: ParamsAny;
   protected rec: TestCaseRecorder;
+  private outstanding = 0;
```
Comment (austinEng) [inline src/framework/fixture.ts]: nit: Know this is a refactor, but maybe "outstandingRejections" or outstandingPromises?

File: src/suites/cts/fences.spec.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+});
+
+// Test initial value.
+g.test('initial/initialValue', t => {
+  const fence = t.queue.createFence({ initialValue: 2 });
+  t.expect(fence.getCompletedValue() === 2);
+});
+
+// Promise resolves when onCompletion value is less than signal value.
+g.test('signal/less', async t => {
```
Comment (austinEng) [inline src/suites/cts/fences.spec.ts]: nit: on completion less ? I think less isn't so descriptive and wouldn't mean much in logs.

File: src/suites/cts/fences.spec.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  const fence = t.queue.createFence({});
+  t.queue.signal(fence, 2);
+  await fence.onCompletion(1);
+  t.expect(fence.getCompletedValue() === 2);
+});
+
+// Promise resolves when onCompletion value is equal to signal value.
+g.test('signal/equal', async t => {
```
Comment (austinEng) [inline src/suites/cts/fences.spec.ts]: ditto

File: src/framework/fixture.ts (hunk 1 above)
Comment (kainino0x) [reply src/framework/fixture.ts]: Good idea, now "numOutstandingAsyncExpectations"

File: src/suites/cts/fences.spec.ts (hunk 2 above)
Comment (kainino0x) [reply src/suites/cts/fences.spec.ts]: Renamed a bunch of the tests and now they're much more consistent.

**Pull request merged**